### ISSUE TITLE
weird results in populated data

### ISSUE THREAD

**Issue by mikermcneil** [balderdashy/waterline#503]
###Details
• Two models in **sails-disk** @ edge (master- so it's skipping the integrator- however I don't think that matters in this case)
• User has a via-less, plural association called `pets` which is a collection of `Pet`

###Steps to reproduce

If I try to do a nested update using primary key values which do not actually belong to any records in the associated model:

```js
error: Sent 500 ("Server Error") response
error: Error (E_UNKNOWN) :: Encountered an unexpected error
    at new WLError (/Users/mikermcneil/code/waterline/lib/waterline/error/WLError.js:33:18)
    at duckType (/Users/mikermcneil/code/waterline/lib/waterline/error/index.js:67:10)
    at errorify (/Users/mikermcneil/code/waterline/lib/waterline/error/index.js:39:10)
    at wrappedCallback (/Users/mikermcneil/code/waterline/lib/waterline/utils/normalize.js:313:15)
    at _normalizeCallback.callback.error (/Users/mikermcneil/code/waterline/node_modules/node-switchback/lib/normalize.js:42:26)
    at _switch (/Users/mikermcneil/code/waterline/node_modules/node-switchback/lib/factory.js:32:25)
    at /Users/mikermcneil/code/waterline/lib/waterline/query/dql/update.js:207:23
    at taskCallback (/Users/mikermcneil/code/waterline/node_modules/async/lib/async.js:456:21)
    at /Users/mikermcneil/code/waterline/lib/waterline/model/lib/defaultMethods/save.js:155:14
    at /Users/mikermcneil/code/waterline/node_modules/async/lib/async.js:439:17

Details:  [ { type: 'insert',
    collection: 'pet_pets_pet__user_pets',
    criteria: { pet_pets_pet: 5, user_pets: 1 },
    values: { pet_pets_pet: 5, user_pets: 1 },
    err: [Error: Associated Record For pet with id = 5 No Longer Exists] },
  { type: 'insert',
    collection: 'pet_pets_pet__user_pets',
    criteria: { pet_pets_pet: 6, user_pets: 1 },
    values: { pet_pets_pet: 6, user_pets: 1 },
    err: [Error: Associated Record For pet with id = 6 No Longer Exists] },
  { type: 'insert',
    collection: 'pet_pets_pet__user_pets',
    criteria: { pet_pets_pet: 7, user_pets: 1 },
    values: { pet_pets_pet: 7, user_pets: 1 },
    err: [Error: Associated Record For pet with id = 7 No Longer Exists] } ]
```

I get a WLError- which is perfect!  But, when I try to find + populate users, the weirdness starts:

```js
User.find().populate('pets').exec(console.log);
// ->
/*
[ { pets: [ {}, {}, {} ],
    name: 'andrew',
    createdAt: '2014-07-01T16:14:08.722Z',
    updatedAt: '2014-07-01T16:16:23.977Z',
    id: 1 },
  { pets: 
     [ { name: 'winget',
         createdAt: '2014-07-01T16:14:14.895Z',
         updatedAt: '2014-07-01T16:14:14.895Z',
         id: 1 },
       { name: 'sparky',
         createdAt: '2014-07-01T16:14:16.911Z',
         updatedAt: '2014-07-01T16:14:16.911Z',
         id: 2 } ],
    name: 'josh',
    createdAt: '2014-07-01T16:14:10.274Z',
    updatedAt: '2014-07-01T16:15:14.600Z',
    id: 2 } ]
*/
```

So.. that's pretty weird by itself.  But stranger still, check out what happens when I peek around inside the junctor relation:

```js
sails.models['pet_pets_pet__user_pets'].find().exec(console.log)
// ->
/*
[ { pet_pets_pet: 2, user_pets: 2, id: 5 },
  { pet_pets_pet: 1, user_pets: 2, id: 6 } ]
*/
```

**Comment by dobaduc** [comment]
I've got exactly the same issue. Could anyone please kindly advise how to fix this?

**Comment by dmarcelino** [comment]
Hi @dobaduc, it looks like this will require code changes in waterline or sails-disk (is this the adapter you're using?). Either way we probably should start by writing an automated test so we have means to reliably reproduce this and confirm any code change has fixed it.

**Comment by dobaduc** [comment]
@dmarcelino Thank you for your prompt response. I am using **sails-mysql** :)

**Comment by dobaduc** [comment]
Please let me attach my error log here for reference

````
 Sending 500 ("Server Error") response: 
  Error (E_UNKNOWN) :: Encountered an unexpected error
     at new WLError (/server/my-api/node_modules/sails/node_modules/waterline/lib/waterline/error/WLError.js:33:18)
     at duckType (/server/my-api/node_modules/sails/node_modules/waterline/lib/waterline/error/index.js:67:10)
     at errorify (/server/my-api/node_modules/sails/node_modules/waterline/lib/waterline/error/index.js:39:10)
     at wrappedCallback (/server/my-api/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:320:15)
     at _normalizeCallback.callback.error (/server/my-api/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/normalize.js:42:31)
     at _switch (/server/my-api/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/factory.js:46:28)
     at /server/my-api/node_modules/sails/node_modules/waterline/lib/waterline/query/dql/update.js:213:23
     at taskCallback (/server/my-api/node_modules/sails/node_modules/async/lib/async.js:471:21)
     at /server/my-api/node_modules/sails/node_modules/waterline/lib/waterline/model/lib/defaultMethods/save.js:167:14
     at /server/my-api/node_modules/sails/node_modules/async/lib/async.js:454:17
     at /server/my-api/node_modules/sails/node_modules/async/lib/async.js:444:17
     at Array.forEach (native)
     at _each (/server/my-api/node_modules/sails/node_modules/async/lib/async.js:46:24)
     at Object.taskComplete (/server/my-api/node_modules/sails/node_modules/async/lib/async.js:443:13)
     at processImmediate [as _immediateCallback] (timers.js:345:15)
 Details:  [ { type: 'insert',
     collection: 'language_users__user_languages',
     criteria: { language_users: 'vi', user_languages: 4 },
     values: { language_users: 'vi', user_languages: 4 },
     err: [Error: Associated Record For language with code = vi No Longer Exists] } ]
````

**Comment by dmarcelino** [comment]
@dobaduc, can you share the relevant model definitions and the code snippet responsible for throwing that error? Thanks!

**Comment by sailsbot** [comment]
Thanks for posting, @mikermcneil.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
